### PR TITLE
refactor(storage): remove the Option of CompactionGroupId in function levels

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

as title

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- remove`Option` of `CompactionGroupId` in function levels
- remove risectl list-kv with None table_id
- fix unit-test

## Checklist

- ~[ ] I have written necessary rustdoc comments~
- ~[ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- https://github.com/risingwavelabs/risingwave/pull/5620

### CODE REVIEW

File: src/storage/src/hummock/state_store.rs [hunk 1]
```diff
@@ -163,81 +160,85 @@ impl HummockStorage {
...
-
-                let pruned_sstables = match T::Direction::direction() {
-                    DirectionEnum::Backward => matched_table_infos.iter().rev().collect_vec(),
-                    DirectionEnum::Forward => matched_table_infos.iter().collect_vec(),
-                };
-
-                let mut sstables = vec![];
-                for sstable_info in pruned_sstables {
-                    if let Some(bloom_filter_key) = prefix_hint.as_ref() {
+        if pinned_version.is_valid() {
```
Comment (wenym1) [inline src/storage/src/hummock/state_store.rs:163]: Seems unnecessary to add this check.

Comment (wenym1) [reply src/storage/src/hummock/state_store.rs:163]: Or we shall `assert!(pinned_version.is_valid())`.

File: src/storage/src/hummock/state_store.rs [hunk 2]
```diff
@@ -338,17 +337,60 @@ impl HummockStorage {
...
         // read/write path.
-        for level in pinned_version.levels(compaction_group_id) {
-            if level.table_infos.is_empty() {
-                continue;
-            }
-            match level.level_type() {
-                LevelType::Overlapping | LevelType::Unspecified => {
-                    let table_infos = prune_ssts(level.table_infos.iter(), &(key..=key));
-                    for table_info in table_infos {
+        if pinned_version.is_valid() {
```
Comment (wenym1) [inline src/storage/src/hummock/state_store.rs:340]: ditto

**Review by soundOfDestiny — APPROVED** [review]
lgtm

**Review by wenym1 — APPROVED** [review]
LGTM!

**Pull request merged**